ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -533,6 +533,10 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) throws Inte
         todo Find a more generic approach to collecting files in the same logical bucket to compact within the same
         task (currently we're using Tez split grouping).
         */
+        if (AcidUtils.isInsertOnlyTable(t.getParameters()) && !HiveConf
+            .getBoolVar(conf, HiveConf.ConfVars.HIVE_COMPACTOR_COMPACT_MM)) {
+          throw new HiveException("Insert only compaction is disabled. Set hive.compactor.compact.insert.only to true to enable it.");
+        }
```

Comment:
Could we put this logic in the QueryCompactorFactory for better readability?

Reply:
done